tc/autotuner/options_cache-inl.h

```diff
@@ -258,6 +285,44 @@ OptionsCache<Backend>::getTopKOptions(
   return res;
 }
 
+} // namespace detail
+
+template <typename Backend>
+std::vector<typename Backend::MappingOptionsType>
+OptionsCache<Backend>::getTopKOptions(
+    const lang::CanonicalTcString& tc,
+    const std::vector<TensorInfo>& inputs,
+    const std::vector<TensorInfo>& outputs,
+    const std::string& backendStr,
+    size_t K) const {
+  OptionsCacheKey key{tc, inputs, outputs, backendStr};
+  return detail::getTopKOptionsImpl<Backend>(
+      mutex,
+      numberAttemptedRetrievals,
+      numberSuccessfulRetrievals,
+      store_,
+      key,
+      K);
+}
+
+template <typename Backend>
+std::vector<typename Backend::MappingOptionsType>
+OptionsCache<Backend>::getTopKOptions(
+    const typename Backend::ExecutorType& exec,
+    size_t K) const {
+  OptionsCacheKey key{canonicalTc(exec.tc()),
+                      exec.inputsInfo(),
+                      exec.outputsInfo(),
+                      exec.deviceName()};
+  return detail::getTopKOptionsImpl<Backend>(
+      mutex,
+      numberAttemptedRetrievals,
+      numberSuccessfulRetrievals,
+      store_,
+      key,
+      K);
```

Comment:
Why do you need a `getTopKOptionsImpl`?
Why can't you directly call the other `getTopKOptions`?

Reply:
good question